### PR TITLE
Initial project

### PR DESCRIPTION
**Pull request by innomaxx**

### CODE REVIEW

File: src/main/java/com/max/grpc/orders/server/CafeServer.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            try {
+                server.awaitTermination();
+            }
+            catch (InterruptedException ex) {
+                logger.fatal("Failed to block runtime until shutdown");
+            }
+        }
+    }
+
+    class MenuServiceImpl extends MenuServiceGrpc.MenuServiceImplBase {
```
Comment (GalynaY) [inline src/main/java/com/max/grpc/orders/server/CafeServer.java]: Move to own file

Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/server/CafeServer.java]: Done

File: src/main/java/com/max/grpc/orders/server/Main.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+package com.max.grpc.orders.server;
+
+import com.max.grpc.orders.server.food.FoodManager;
+
+public class Main {
+    public static void main(String[] args) {
+        var foodManager = new FoodManager();
+        var server = new CafeServer(8000, foodManager);
```
Comment (malloycv) [inline src/main/java/com/max/grpc/orders/server/Main.java]: What is 8000 mean?

File: src/main/java/com/max/grpc/orders/server/services/MenuServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public MenuServiceImpl(FoodManager foodManager) {
+        this.foodManager = foodManager;
+    }
+
+    @Override
+    public void getMenu(Empty request, StreamObserver<CafeMenu> responseObserver) {
+        var cafeMenuBuilder = CafeMenu.newBuilder();
+
+        FoodItem[] menu = foodManager.getMenu();
+        for (FoodItem foodItem : menu) {
```
Comment (malloycv) [inline src/main/java/com/max/grpc/orders/server/services/MenuServiceImpl.java]: Do you know how to use Stream API ?

File: src/main/java/com/max/grpc/orders/server/services/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        for (String itemId : foodIds) {
+            FoodItem item = foodManager.getItemById(itemId);
+            if (item != null) {
+                chosenItems.add(item);
+            }
+        }
+
+        var receiptBuilder = OrderReceipt.newBuilder();
+        receiptBuilder.setId(UUID.randomUUID().toString());
+        receiptBuilder.setDate((long)(System.currentTimeMillis() / 1e3));
```
Comment (malloycv) [inline src/main/java/com/max/grpc/orders/server/services/OrderServiceImpl.java:36]: what is 1e3 here ?

File: src/main/java/com/max/grpc/orders/server/Main.java (hunk 2 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/server/Main.java]: Port

Comment (malloycv) [reply src/main/java/com/max/grpc/orders/server/Main.java]: Oh, I see, thanks for the explanation. But the code should be descriptive enough, to do not ask developers about such things.

Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/server/Main.java]: Ok, please suggest how to make this code more descriptive? Use builder? Or explicitly declare local variable and pass it?

Comment (malloycv) [reply src/main/java/com/max/grpc/orders/server/Main.java]: You can make a constant and call it SERVER_PORT, and simply use it. Or even more, you can extract it to the property files, to make your server more configurable.

File: src/main/java/com/max/grpc/orders/server/services/MenuServiceImpl.java (hunk 3 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/server/services/MenuServiceImpl.java]: Sure, I'll rewrite this loop

File: src/main/java/com/max/grpc/orders/server/services/OrderServiceImpl.java (hunk 4 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/server/services/OrderServiceImpl.java:36]: 1000

File: src/main/java/com/max/grpc/orders/server/services/MenuServiceImpl.java (hunk 3 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/server/services/MenuServiceImpl.java]: I found much simpler approach than using Stream API

File: src/main/java/com/max/grpc/orders/server/services/OrderServiceImpl.java (hunk 4 above)
Comment (malloycv) [reply src/main/java/com/max/grpc/orders/server/services/OrderServiceImpl.java:36]: This is rhetorical question. Here the same problem with magic number.

File: src/main/java/com/max/grpc/orders/server/Main.java (hunk 2 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/server/Main.java]: Added loading server port from `resources/server.properties`

File: src/main/java/com/max/grpc/orders/server/services/OrderServiceImpl.java (hunk 4 above)
Comment (innomaxx) [reply src/main/java/com/max/grpc/orders/server/services/OrderServiceImpl.java:36]: Removed inappropriate usage of `1e3` and now date calculation is more descriptive

**Pull request merged**